File: {{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/mutations.py
# add GraphQL mutations here
import graphene
from django.contrib.auth import get_user_model

from .types import UserType


class UserInput(graphene.InputObjectType):
    email = graphene.String(required=True)
    password = graphene.String(required=True)
    first_name = graphene.String(required=True)
    last_name = graphene.String(required=True)


class CreateUser(graphene.Mutation):
    class Arguments:
        data = UserInput(required=True)

    user = graphene.Field(UserType)

    def mutate(self, info, data):
        user = get_user_model().objects.create_user(**data)
        return CreateUser(user=user)

File: {{cookiecutter.project_slug}}/clients/react/src/utils/mutations.ts
import gql from "graphql-tag";

/* add GraphQL client-side mutations here */

// Auth mutations
export const LOG_IN = gql`
  mutation logIn($email: String!, $password: String!) {
    tokenAuth(email: $email, password: $password) {
      token
    }
  }
`;

export const REFRESH_TOKEN = gql`
  mutation RefreshToken($token: String!) {
    refreshToken(token: $token) {
      token
      payload
      refreshExpiresIn
    }
  }
`;

export const VERIFY_TOKEN = gql`
  mutation VerifyToken($token: String!) {
    verifyToken(token: $token) {
      payload
    }
  }
`;


export const CREATE_USER = gql`
  mutation CreateUser($data: UserInput!) {
    createUser(data: $data) {
      user {
        id
        email
      }
    }
  }
`
